Sources/MapboxMaps/Foundation/Extensions/Core/ResourceOptions.swift

```diff
@@ -63,4 +71,38 @@ extension ResourceOptions {
 
         return cacheDirectoryURL.appendingPathComponent("cache.db")
     }
+
+    /// :nodoc:
+    public override func isEqual(_ object: Any?) -> Bool {
+
+        guard let other = object as? ResourceOptions else {
+            return false
+        }
+
+        guard type(of: self) == type(of: other) else {
+            return false
+        }
+
+        return
+            (accessToken == other.accessToken) &&
```

Comment:
Can be addressed later

```suggestion
        return
            (type(of: self) == type(of: other)) &&
            (accessToken == other.accessToken) &&
```